src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala

```diff
@@ -242,7 +243,25 @@ class GenotypeRecordReader(vcfSettings: VCFSettings) extends HtsjdkRecordReader[
   }
 }
 
-class GenericRecordReader extends HtsjdkRecordReader[Annotation] {
+
+object GenericRecordReader {
+  val haploidRegex = """^([0-9]+)$""".r
+  val diploidRegex = """^([0-9]+)([|\/]([0-9]+))$""".r
```

Comment:
It shouldn't be necessary.  You're escaping a / which doesn't need to be escaped.

scala> val r = """[0-9]+[|/][0-9]+""".r
r: scala.util.matching.Regex = [0-9]+[/|][0-9]+

scala> r.findFirstIn("01")
res8: Option[String] = None

scala> r.findFirstIn("0/1")
res5: Option[String] = Some(0/1)

scala> r.findFirstIn("2|2")
res7: Option[String] = Some(2|2)

You're just building a string here, so you can do s"...${VCFConstants.UNPHASED}...".r although honestly I think  it is clearer hardcoded.